### ISSUE TITLE
create mutation fails when owner field is not set - initiated from amplify-ios datastore

### ISSUE THREAD

**Issue by onlybakam** [aws-amplify/amplify-ios#303]
**Describe the bug**
the issue is opened for the CLI: https://github.com/aws-amplify/amplify-cli/issues/3291

Adding it here because this problems happens with amplify-ios models and the way the input string is built.

**Comment by lawmicha** [comment]
Hi @onlybakam, thanks for opening this issue, can you clarify what is your schema and how you are making calls with the generated models? Are you using `Amplify.DataStore.save` or `Amplify.API.mutate` with the graphql request builders given the model? From there we can see if the issue still exists.

There is similar problem that was fixed in Amplify iOS library. When using auth directive with owner, the models were being generated with default parameters, for example:
```
type Model @model @auth(allow: "owner") {
 id: ID!
 owner: Sring
}
```
When creating an instance of Model and transforming that into a graphql request, the owner field was populated with `null` in the query variables. There is a client side fix that patches this by filtering out owner field based on the auth directive done in https://github.com/aws-amplify/amplify-ios/pull/423. So we should still keep the CLI issue open, while maintaining this client side patch.

**Comment by onlybakam** [comment]
hey @lawmicha what you're describing as fixed in #423 is the issue i was having: 'remove ownerField from variables when is nil'. so this should have solved it.

**Comment by lawmicha** [comment]
this issue was to track [CLI issue](https://github.com/aws-amplify/amplify-cli/issues/3291). But now that I think about this more, if there is an resolution from the CLI, the patch on the client side cannot be removed for clients to be backwards compatiable. Developer would have to reprovision their backend to pick up the GraphQL Transform changes. Since there isn't a change to be done on the client side, i will close this issue as mentioned by @onlybakam, the problem was fixed in #423 . Feel free to re-open if this is not the case
